### ISSUE TITLE
object-curly-spacing treats trailing computed member access as array

### ISSUE THREAD

**Issue by ksmithut** [eslint/eslint#5634]
**What version of ESLint are you using?**
2.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{
  "rules": {
    "object-curly-spacing": [ 2, "always", {
      "objectsInObjects": false,
      "arraysInObjects": false
    }],
    "array-bracket-spacing": [ 2, "always", {
      "singleValue": false,
      "arraysInArrays": false,
      "objectsInArrays": false
    }]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
var data = { foo: 'bar' }
var key = 'foo'

var obj = { prop: data[foo] } // Error on this line
var arr = [ 1, data[foo] ] // None on this line
```

**What did you expect to happen?**
I expected there not to be an error because my rules say that I need a space padding in objects, unless arrays or objects on on either edge of the padding. In the above example, I don't have an object or array on the edge of the object.

**What actually happened? Please include the actual, raw output from ESLint.**
>   4:29  error  There should be no space before '}'  object-curly-spacing
> 
> ✖ 1 problem (1 error, 0 warnings)

It appears that the rule for `arraysInObjects` is just checking for a square bracket instead of seeing if it's an array. It appears to be checking it correctly with the `array-bracket-spacing` rule

**Comment by michaelficarra** [comment]
Yeah, it looks like a bug. Please use a more descriptive issue title next time. I would have used

> object-curly-spacing treats trailing computed member access as array

**Comment by ksmithut** [comment]
Good catch, I meant to change it before I submitted. Thanks

**Comment by BYK** [comment]
Duplicate of #5538 which is fixed on master so closing. Reopen if you disagree please :)
